src/js/components/editor/Editor.vue

```diff
@@ -0,0 +1,284 @@
+<template>
+    <div class="text-pane" :class='{fullscreen}'>
+        <toolbar :state="state"
+            @fullscreen="toggleFullScreen"
+        />
+        <div v-if="!colInRouter">
+            <text-selector @selectedColumn="getText"></text-selector>
+        </div>
+        <section class="editor">
+            <div class="editor-column" v-for="(column, columnIndex) of currentText.cols" :data-col="column.col" :key="columnIndex + column.col">
+                <input id="editor-input" class="" @keyup="onKeyup" @blur="onInputBlur" v-model="input" />
+                <table class="editor-table" >
+                    <thead>
+                        <th>{{column.col}}</th>
+                        <th class="line-col"></th>
+                    </thead>
+                    <tbody>
+                        <tr v-for="(line, lineIDX) of column.lines" :key="lineIDX + line.lineId" :data-line="line.line" :data-line-id="line.lineId"
+                            dir="rtl">
+                            <td class="text-col">
+                                <sqe-sign v-for="sign of line.signs"
+                                    @click="onClick"
+                                    :sign="sign"
+                                    :state="state" 
+                                    :focusedSignId="focusedSignId"
+                                    :key="sign.id"
+                                />
+                            </td>
+                            <td class="line-col">{{line.line}}</td>
+                        </tr>
+                    </tbody>
+                </table>
+            </div>
+        </section>
+    </div>
+</template>
+
+<script>
+
+import KEY_CODES from './key_codes.js';
+import TextSelector from './TextSelector.vue'
+import Toolbar from './Toolbar.vue'
+import SignStreamProcessor from '~/utils/SignStreamProcessor.js'
+import Sign from './Sign.vue';
+import editorStore from './EditorStore.js'
+
+export default {
+    components: {
+        'sqe-sign': Sign,
+        'text-selector': TextSelector,
+        'toolbar': Toolbar
+    },
+    computed: {
+
+        /**
+         * @type {string} the sign id for the currently focused sign 
+         */
+        focusedSignId() {
+            return this.focusedSign ? this.focusedSign.id : -1
+        },
+
+        colInRouter() {
+            return (this.$route.params.colID !== '~' && this.$route.params.colID > -1)
```

Comment:
I just checked the database and you should use 0 instead of -1.  The primary keys start at 1, not 0.

Reply:
Thanks, patched these up.